Source/GSCSVariable.m

```diff
@@ -24,4 +24,146 @@
 
 @implementation GSCSVariable
 
+- (instancetype) init
+{
+  self = [super init];
+  if (self != nil)
+    {
+      _type = GSCSVaraibleTypeVariable;
+    }
+
+  return self;
+}
+
+- (instancetype) initWithName: (NSString *)name
+{
+  self = [super init];
+  if (self != nil)
+    {
+      ASSIGN(_name, name);
+      _type = GSCSVaraibleTypeVariable;
+    }
+
+  return self;
+}
+
+- (instancetype) initWithName: (NSString *)name type: (GSCSVariableType)type
+{
+  self = [super init];
+  if (self != nil)
+    {
+      ASSIGN(_name, name);
+      _type = type;
+    }
+
+  return self;
+}
+
++ (instancetype) dummyVariableWithName: (NSString *)name
+{
+  return AUTORELEASE([[GSCSVariable alloc] initWithName: name type: GSCSVariableTypeDummy]);
+}
+
++ (instancetype) slackVariableWithName: (NSString *)name
+{
+  return AUTORELEASE([[GSCSVariable alloc] initWithName: name type: GSCSVariableTypeSlack]);
+}
+
++ (instancetype) objectiveVariableWithName: (NSString *)name
+{
+  return AUTORELEASE([[GSCSVariable alloc] initWithName: name
+                                       type: GSCSVariableTypeObjective]);
+}
+
++ (instancetype) variableWithValue: (CGFloat)value
+{
+  return [self variableWithValue: value name: nil];
+}
+
++ (instancetype) variable
+{
+  return [self variableWithValue: 0 name: nil];
+}
+
++ (instancetype) variableWithValue: (CGFloat)value name: (NSString *)name
+{
+  GSCSVariable *variable =
+      [[GSCSVariable alloc] initWithName: name type: GSCSVariableTypeExternal];
+  [variable setValue: value];
+  return AUTORELEASE(variable);
+}
+
+- (BOOL) isDummy
+{
+  return _type == GSCSVariableTypeDummy;
+}
+
+- (BOOL) isExternal
+{
+  return _type == GSCSVariableTypeExternal;
+}
+
+- (BOOL) isPivotable
+{
+  return _type == GSCSVariableTypeSlack;
+}
+
+- (BOOL) isRestricted
+{
+  return _type == GSCSVariableTypeExternal || _type == GSCSVariableTypeSlack;
+}
+
+- (NSString *) description
+{
+  if (_type == GSCSVariableTypeDummy)
+    {
+      return [NSString stringWithFormat: @"[%@:dummy]", _name];
+    }
+  else if (_type == GSCSVariableTypeSlack)
+    {
+      return [NSString stringWithFormat: @"[%@:slack]", _name];
+    }
+  else if (_type == GSCSVariableTypeExternal)
+    {
+      return [NSString stringWithFormat: @"[%@:%.02f]", _name, _value];
+    }
+  else if (_type == GSCSVariableTypeObjective)
+    {
+      return [NSString stringWithFormat: @"[%@:objective]", _name];
+    }
+
+  return [NSString stringWithFormat: @"[%@]", _name];
+}
+
+- (GSCSVariableType) type
+{
+  return _type;
+}
+
+- (NSUInteger) id
+{
+  return _id;
+}
+
+- (CGFloat) value
+{
+  return _value;
+}
+
+- (void) setValue: (CGFloat)value
+{
+  _value = value;
```

Comment:
If setting the value is possible for all types of variables, why isn't the value displayed in the -description method?

Reply:
This sounds like the best way to go.